src/systems/physics/Physics.cc

```diff
@@ -3811,9 +3824,36 @@ void PhysicsPrivate::UpdateCollisions(EntityComponentManager &_ecm)
           for (const auto &contact : contactData)
           {
             auto *position = contactMsg->add_position();
-            position->set_x(contact->point.x());
-            position->set_y(contact->point.y());
-            position->set_z(contact->point.z());
+            position->set_x(contact.first->point.x());
+            position->set_y(contact.first->point.y());
+            position->set_z(contact.first->point.z());
+
+            // Check if the extra contact data exists,
+            // since not all physics engines support it.
+            // Then, fill the msg with extra data.
+            if(contact.second != nullptr)
+            {
+              auto *normal = contactMsg->add_normal();
+              normal->set_x(contact.second->normal.x());
+              normal->set_y(contact.second->normal.y());
+              normal->set_z(contact.second->normal.z());
+
+              auto *wrench = contactMsg->add_wrench();
+              auto *body1_wrench = wrench->mutable_body_1_wrench();
```

Comment:
nit: please use `camelCase` for `body1_wrench`, `body1_force`, `body2_wrench`, and `body2_force`.